lib/SPIRV/SPIRVReader.cpp

```diff
@@ -562,12 +563,32 @@ void SPIRVToLLVM::setLLVMLoopMetadata(SPIRVLoopMerge *LM, BranchInst *BI) {
     Name = llvm::MDString::get(*Context, "llvm.loop.unroll.full");
   else if (LM->getLoopControl() == LoopControlDontUnrollMask)
     Name = llvm::MDString::get(*Context, "llvm.loop.unroll.disable");
-  else
+  else if (LM->getLoopControl() == LoopControlDependencyInfiniteMask)
+    Name = llvm::MDString::get(*Context, "llvm.loop.ivdep.enable");
+  else if (LM->getLoopControl() == LoopControlDependencyLengthMask) {
+    LoopControlParameters = LM->getLoopControlParameters();
+    Name = llvm::MDString::get(*Context, "llvm.loop.ivdep.safelen");
+  } else if (LM->getLoopControl() == LoopControlExtendedControlsMask) {
+    LoopControlParameters = LM->getLoopControlParameters();
+    Name =
+        (LoopControlParameters.front() == InitiationIntervalINTEL)
+            ? llvm::MDString::get(*Context, "llvm.loop.ii.count")
+            : llvm::MDString::get(*Context, "llvm.loop.max_concurrency.count");
+
+    // Keep only last parameter in metadata
```

Comment:
Why?

Reply:
Because the loop control mask is being shared between two attributes (and therefore LLVM metadatas). What metadata to chose is decided by picking an appropriate parameter set for the mask. We don't want to see the parameter in metadata generated. Please consider following IR:
1. Without erasing:
"llvm.loop.ii.count", i32 0x800000, i32 /*count*/     // incorrect
2. With erasing:
"llvm.loop.ii.count", i32 /*count*/                            // correct